### ISSUE TITLE
In firefox and edge, move the mouse to select the text

### ISSUE THREAD

**Issue by DreamLi1314** [codemirror/CodeMirror#5370]
In firefox and edge, moving the mouse will selects the text, which looks the same as moving the mouse with the left mouse button. 

I am using CodeMirror error in angular2.

**Comment by marijnh** [comment]
This is not generally the case, and doesn't appear to happen on the demos on the website, so I'm going to need some more information and/or a minimal setup that shows the problem.

**Comment by DreamLi1314** [comment]
Thanks, I really should be so.

**Comment by DreamLi1314** [comment]
I found this problem. because in CodeMirror/src/util/event.js, the e_button method tries to get the current mouse button using e.which, but when the mouse is not pressed the which property gets 1 in Firefox, get in Chrome is 0, so firefox browsers will think that the left mouse button was pressed.
I think this method seems to return e.buttons directly.
I reproduce this issue at least on firefox and edge, please check this issue and other browsers.

**Comment by DreamLi1314** [comment]
I temporarily choose to listen for the MouseMove event and filter it to avoid this problem.
as follows:
`
mouseMove(event: MouseEvent): void {
      if(event.buttons != 1) { 
         event.stopPropagation(); 
         event.preventDefault(); 
      }
   }
`

**Comment by marijnh** [comment]
I agree that using `buttons` here when supported is a good idea. In general `mouseup` is reliable on Firefox, so I'm curious why that isn't firing in your situation. Does attached patch fix the problem?

**Comment by DreamLi1314** [comment]
My problem is caused by mouseEvent.which is 1 on firefox when the mouse is moved without pressing the left key.
I successfully solved my problem using attached patch.
